### ISSUE TITLE
Sleep Delay being Ignored, throwing Inaccurate Error

### ISSUE THREAD

**Issue by gwergoth** [InstaPy/InstaPy#2038]
## Expected Behavior

A sleep delay every 10 follows, as per the sleep_delay setting, when running follow_user_followers.

## Current Behavior

The bot doesn't sleep. I'm not getting the "sleeping for about Xmin" message at all. It skips on to the next username in the list and throws an error. This error is however incorrect. The user does indeed have enough followers to follow and the bot then goes on and starts following them.

This seemed to just start happening out of nowhere. The only thing I changed was merging all of my csv log files after multi_logs was set to true and they started being written to a subfolder. But I'm pretty sure it was working as expected the last time I ran it and now it's playing up again. I'm running the latest version of Instapy.

From my logs...

INFO [2018-05-17 12:48:04] [myaccount]  --> Ongoing follow 8, now following: user489
INFO [2018-05-17 12:48:28] [myaccount]  --> Ongoing follow 9, now following: user3b5
INFO [2018-05-17 12:48:47] [myaccount]  --> Ongoing follow 10, now following: user93d
WARNING [2018-05-17 12:49:47] [myaccount]  _art_psycho -> Less users to follow than requested.
INFO [2018-05-17 12:49:53] [myaccount]  --> Ongoing follow 1, now following: user6ac
INFO [2018-05-17 12:50:15] [myaccount]  --> Ongoing follow 2, now following: user4bb
INFO [2018-05-17 12:50:34] [myaccount]  --> Ongoing follow 3, now following: user996

## InstaPy configuration

    session.set_relationship_bounds(enabled=False,
				 potency_ratio=-1.20,
				  delimit_by_numbers=False,
				   max_followers=4590,
				    max_following=5555,
				     min_followers=45,
				      min_following=77)
				      
    session.set_do_comment(True, percentage=20)
    session.set_comments([u'\uD83D\uDC4D\uD83C\uDFFB So wow!', 'Such nice!', 'Many amazing!'])

    session.like_by_tags(['tag1', 'tag2', 'tag3', 'tag4', 'tag5','tag6'], amount=6, media='Photo', skip_top_posts=False)

    session.follow_user_followers(['user1','user2','user3','user4', 'user5'], amount=10, sleep_delay=555, randomize=True, interact=False)

    session.unfollow_users(amount=60, onlyInstapyFollowed = True, onlyInstapyMethod = 'FIFO', sleep_delay=555, unfollow_after=48*60*60)

**Comment by uluQulu** [comment]
Yes @grrenth, it's because the break time is handled per USER basis.
It will sleep if you follow more than 10 usernames per USERs in your list, but if less, it will not sleep and continue with the USER (_if any_).
It is an unwanted behavior and with my latest PR- #2012 targeted mainly `follow_user_followers` and `follow_user following` features, also comes what you need- break time handling per follow feature-wide rather than per follow per USER.
>_Here USER is an element from the given list `['user1','user2','user3','user4', 'user5']`_

**Comment by gwergoth** [comment]
Aha! Thanks for shedding some light on that @uluQulu. I just couldn't figure out why it only started happening and then only sometimes.

Now I just hope that I haven't inadvertently raised any flags. I usually run the bot three or four times a day, following no more than 60 users each run, and I set the sleep delay so that I'm never following them any faster than within an hour.

But looking through my logs, there's at least a couple of times when I've followed up to 90 in around 45 minutes. Do we have any sort of realistic idea of current Instagram follow and like limits? I really want to fly under the radar and hope this isn't going to have jeopardised that.

**Comment by uluQulu** [comment]
@grrenth,
A few minutes back #2012 was merged. Feel free to pull the latest files to use the improvements coming with it!
I have no accurate information and so can't say about that.  
_But once I wrote a realtime inspector module to take InstaPy under full control, named [Quota Supervisor](https://github.com/uluQulu/InstaPy-with-QS#quota-supervisor), which could help you to not get caught by radar, although it is discontinued now.._
